### PR TITLE
Feat/see response code

### PR DESCRIPTION
**Pull request by PeterNgTr**
Fulfilled https://github.com/Codeception/CodeceptJS/issues/244

### CODE REVIEW

**Comment by PeterNgTr** [comment]
Updated code as grabbing the statusCode of current URL and make assertion against it.

**Comment by PeterNgTr** [comment]
Fixed failed helper tests

**Comment by DavertMik** [comment]
Sorry, still no. 

Here is the thing. When you send requests via 3rd party library (which is got) you are not inside the session of current browser so the results will be different from what you get in browser. 

How I think it should be implemented

* Nightmare has API for status codes: https://github.com/segmentio/nightmare/issues/656
* Puppeteer has its API too.
* WebDriver/Protractor doesn't have such API at all.

In Nightmare/Puppeteer you can just create a property for a status code and update it after each request. When you call `seeResponseCodeIs` it will compare expected value with the value of the last request.

**Comment by PeterNgTr** [comment]
I‘m trying to implement this with your suggestions

**Comment by PeterNgTr** [comment]
Updated function `seeResponseCodeIs`, right now, after doing a `amOnPage`, the `statusCode` will be updated and `seeResponseCodeIs` will make the assertion against this `statusCode`

File: lib/helper/WebDriverIO.js [hunk 1]
```diff
@@ -773,6 +774,14 @@ class WebDriverIO extends Helper {
     return stringIncludes('web page title').assert(text, title);
   }
 
+  /**
+   * {{> ../webapi/seeResponseCodeIs }}
+   */
+  async seeResponseCodeIs(expectedStatusCode) {
```
Comment (DavertMik) [inline lib/helper/WebDriverIO.js]: WebDriver can't return status codes

File: lib/helper/WebDriver.js [hunk 2]
```diff
@@ -818,6 +819,14 @@ class WebDriver extends Helper {
     return stringIncludes('web page title').assert(text, title);
   }
 
+  /**
+   * {{> ../webapi/seeResponseCodeIs }}
+   */
+  async seeResponseCodeIs(expectedStatusCode) {
+    const statusCode = await getStatusCode(await this.grabCurrentUrl());
```
Comment (DavertMik) [inline lib/helper/WebDriver.js]: WebDriver can't return status codes.
Please remove this method from WebDriver

File: lib/helper/Puppeteer.js [hunk 3]
```diff
@@ -466,10 +467,18 @@ class Puppeteer extends Helper {
...
+    const response = await this.page.goto(url, { waitUntil: this.options.waitForNavigation });
+    statusCode = response.status();
     return this._waitForAction();
   }
 
+  /**
+   * {{> ../webapi/seeResponseCodeIs }}
+   */
+  async seeResponseCodeIs(expectedStatusCode) {
+    return statusCode === expectedStatusCode;
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: this should be an assertion

File: lib/helper/Nightmare.js [hunk 4]
```diff
@@ -412,9 +413,17 @@ class Nightmare extends Helper {
...
       this.debugSection('Headers', JSON.stringify(res.headers));
+      statusCode = res.code;
     });
   }
 
+  /**
+   * {{> ../webapi/seeResponseCodeIs }}
+   */
+  async seeResponseCodeIs(expectedStatusCode) {
+    return statusCode === expectedStatusCode;
```
Comment (DavertMik) [inline lib/helper/Nightmare.js]: should be assertion!

File: lib/helper/WebDriverIO.js [hunk 5]
```diff
@@ -13,6 +13,7 @@ const {
   chunkArray,
   convertCssPropertiesToCamelCase,
   screenshotOutputFolder,
+  getStatusCode,
```
Comment (DavertMik) [inline lib/helper/WebDriverIO.js]: what does this function do?
I don't see it in this PR

**Comment by PeterNgTr** [comment]
Found another issue in test, the `statusCode` will not be set if we are not `await I.amOnPage()`

**Comment by PeterNgTr** [comment]
I'm not sure if we really need this anymore. So closed this

**Pull request closed without merging**